dotnet/src/Connectors/Connectors.Onnx.UnitTests/BertOnnxTextEmbeddingGeneration.cs

```diff
@@ -0,0 +1,383 @@
+                    "Frogs, toads, and salamanders are all examples.",
+                    "A frog is green.",
+                    "Cos'è un anfibio?",
+                    "They are four-limbed and ectothermic vertebrates.",
+                    "A dog is a mammal.",
+                    "A tree is green.",
+                    "It's not easy bein' green.",
+                    "A dog is a man's best friend.",
+                    "You ain't never had a friend like me.",
+                    "Rachel, Monica, Phoebe, Joey, Chandler, Ross",
+                },
+                sortedExamples);
+        }
+    }
+
+    private static void AssertEqualTolerance(ReadOnlySpan<float> left, ReadOnlySpan<float> right)
+    {
+        var cosineSimilarity = TensorPrimitives.CosineSimilarity(left, right);
+        Assert.True(cosineSimilarity >= 0.999995f, $"Expected {cosineSimilarity} >= 0.999995f");
+    }
+
+    private static async Task<string> GetTestFilePath(string url)
+    {
+        // Rather than downloading each model on each use, try to cache it into a temporary file.
+        // The file's name is computed as a hash of the url.
+
+        string name = Convert.ToHexString(SHA256.HashData(Encoding.UTF8.GetBytes(url))) + ".cachedtestfile";
+        string path = Path.Join(Path.GetTempPath(), name);
+
+        if (!File.Exists(path))
+        {
+            using HttpClient client = new();
+            using Stream responseStream = await client.GetStreamAsync(url);
+            using FileStream dest = File.OpenWrite(path);
+            await responseStream.CopyToAsync(dest);
+        }
+
+        return path;
+    }
+
+    private const string BgeMicroV2ModelUrl = "https://huggingface.co/TaylorAI/bge-micro-v2/resolve/main/onnx/model.onnx";
+    private const string BgeMicroV2VocabUrl = "https://huggingface.co/TaylorAI/bge-micro-v2/raw/main/vocab.txt";
+
+    private static async Task<BertOnnxTextEmbeddingGenerationService> GetBgeMicroV2ServiceAsync() =>
+        await BertOnnxTextEmbeddingGenerationService.CreateAsync(
+            await GetTestFilePath(BgeMicroV2ModelUrl),
+            await GetTestFilePath(BgeMicroV2VocabUrl));
+
+    private static async Task<BertOnnxTextEmbeddingGenerationService> GetAllMiniLML6V2Async() =>
+        await BertOnnxTextEmbeddingGenerationService.CreateAsync(
+            await GetTestFilePath("https://huggingface.co/optimum/all-MiniLM-L6-v2/resolve/main/model.onnx"),
+            await GetTestFilePath("https://huggingface.co/optimum/all-MiniLM-L6-v2/raw/main/vocab.txt"));
+
+    private static async Task<BertOnnxTextEmbeddingGenerationService> GetE5BaseV2Async() =>
+        await BertOnnxTextEmbeddingGenerationService.CreateAsync(
+            await GetTestFilePath("https://huggingface.co/intfloat/e5-base-v2/resolve/main/onnx/model.onnx"),
+            await GetTestFilePath("https://huggingface.co/intfloat/e5-base-v2/raw/main/onnx/vocab.txt"));
```

Comment:
Referencing these external repos introduces a bit of risk:

1. Since these reference `main`, any updates to these models may change the behavior and make the tests start failing (since they assert numerical values for the embeddings)
2. The 3rd-party owners of these models might even delete them completely

We could mitigate the first of these easily by changing the URLs to reference an exact commit hash (example: https://huggingface.co/intfloat/e5-base-v2/resolve/1c644c9/onnx/model.onnx).

Mitigating the 2nd would require either:

 * Getting the model authors to add a [DOI](https://huggingface.co/docs/hub/en/doi) (digital object identifier - an immutable, undeletable version of the model)
 * Or just forking the model repo, which is what SmartComponents.LocalEmbeddings does for its default model

Not sure if model deletion is a realistic risk you want to mitigate, but specifying a commit hash at least is probably worthwhile.

Reply:
Thanks. I've added commit hashes.